doc/internal-api.rst

```diff
@@ -66,19 +67,52 @@ methods:
 
 
 
+About the IO module
+-------------------
+
+Handling of the different file formats is implemented in :mod:`can.io`.
+Each file/IO type is within a separate module and ideally implements both a *Reader* and a *Writer*.
+The reader usually extends :class:`can.io.generic.BaseIOHandler`, while
+the writer additionally extends :class:`can.Listener`,
+to be able to be passed directly to a :class:`can.Notifier`.
+
+
+
+Adding support for new file formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This assumes that you want to add a new file format, called *canstore*.
+
+1. Create a new module: *can/io/canstore.py*
+   (*or* simply copy some existing one like *can/io/csv.py*)
+2. Implement a reader ``CanstoreReader`` (which extends :class:`can.io.generic.BaseIOHandler`).
+   Besides from a constructor, only ``__iter__(self)`` needs to be implemented.
+3. Implement a writer ``CanstoreWriter`` (which extends :class:`can.io.generic.BaseIOHandler` and :class:`can.Listener`).
```

Comment:
Again they are optional base classes. The only requirement is the writer must have a `on_message_received` method.